core/range_scan_options.hxx

```diff
@@ -53,11 +59,11 @@ struct range_snapshot_requirements {
 };
 
 struct range_scan_create_options {
-    std::chrono::milliseconds timeout;
     std::string collection_name;
     std::string scope_name;
-    std::uint32_t collection_id{ 0 };
     std::variant<range_scan, sampling_scan> scan_type;
```

Comment:
Could the scan_type be std::variant<std::monostate, range_scan, sampling_scan>?  This would allow the range_scan_create_options to be default-constructible as well, right?

The node autogen relies objects being default-constructible.